fplll/pruner.cpp

```diff
@@ -289,6 +298,11 @@ inline FT Pruner<FT>::single_enum_cost(/*i*/ const evec &b, vector<double> *deta
     total += tmp;
     normalized_radius_pow *= normalized_radius;
   }
+  if (!total.is_finite())
+  {
+    throw std::runtime_error("NaN or inf in single_enum_cost");
```

Comment:
I was thinking that `is_finite` checks if the value is within representable range.